src/test/java/org/reaktivity/nukleus/tcp/internal/streams/ServerPartialWriteIT.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.tcp.internal.streams;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.rules.RuleChain.outerRule;
+import static org.reaktivity.nukleus.tcp.internal.writer.stream.StreamFactory.WRITE_SPIN_COUNT;
+
+import java.io.InputStream;
+import java.net.Socket;
+import java.util.List;
+
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.reaktivity.reaktor.test.NukleusRule;
+
+/**
+ * This test verifies the handling of incomplete writes, when attempts to write data to a socket channel
+ * fail to write out all of the data. In real life this would happen when a client is reading data at a lower
+ * speed than it is being written by the server. For testing purposes this test simulates the condition
+ * by rewriting the bytecode of the SocketChannelImpl.write method to make that method exhibit the behavior of
+ * incomplete writes.
+ */
+@RunWith(org.jboss.byteman.contrib.bmunit.BMUnitRunner.class)
+public class ServerPartialWriteIT
+{
+    private final K3poRule k3po = new K3poRule()
+        .addScriptRoot("route", "org/reaktivity/specification/nukleus/tcp/control/route")
+        .addScriptRoot("streams", "org/reaktivity/specification/nukleus/tcp/streams");
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(5, SECONDS));
+
+    private final NukleusRule nukleus = new NukleusRule("tcp")
+        .directory("target/nukleus-itests")
+        .commandBufferCapacity(1024)
+        .responseBufferCapacity(1024)
+        .counterValuesBufferCapacity(1024)
+        .streams("tcp", "target");
+
+    @Rule
+    public final TestRule chain = outerRule(nukleus).around(k3po).around(timeout);
+
+    @Before
+    public void initWriteHelper()
+    {
+        PartialWriteBytemanHelper.initialize();
```

Comment:
Consider switching this `Helper` to a `TestRule` to enforce re-initialization before each test.